src/slick.grid.ts

```diff
@@ -518,7 +521,12 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
     }
 
     // calculate these only once and share between grid instances
-    this._options = Utils.extend(true, {}, this._defaults, this.options);
+    if (this.options.mixinDefaults) {
```

Comment:
after turning the option to optional as I requested, you'll have to update this if condition, you can use optional chaining `?.` for that in ES2020 (see [MDN: Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) for more info)

~Also for the next line, I think you could simplify the 2 lines with a 1 liner by using Nullish coalescing operator `??` (see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing))~

```ts
// this 1 liner should work since you are reassigning to `this._options` anyway
this._options = Utils.applyDefaults(this.options ?? {}, this._defaults);

// THE PREVIOUS LINE IS IS ACTUALLY INVALID
```

Actually wait, your `applyDefaults` function does not return anything, so your code is actually wrong here and would basically now assign `undefined` to your `this._options` which is bad. I'm surprised that the editor didn't warn you, that's the beauty of TypeScript you should get a lot more warnings and errors of such potential problems before running it.